### PR TITLE
share/p2p: move eds sub

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Self-explanatory

## Checklist

- [x] Required CI checks are passing

### CODE REVIEW

**Review by distractedm1nd — CHANGES_REQUESTED** [review]
shrexpush not sharexpush

File: share/p2p/shrexpush/pubsub.go [hunk 1]
```diff
@@ -1,16 +1,20 @@
-package eds
+package push
```
Comment (renaynay) [inline share/p2p/shrexpush/pubsub.go:1]: shouldn't this be `shrexpush`

Comment (vgonkivs) [reply share/p2p/shrexpush/pubsub.go:1]: ok

**Review by distractedm1nd — APPROVED** [review]

File: share/p2p/shrexpush/pubsub.go [hunk 2]
```diff
@@ -77,7 +81,7 @@ func (s *PubSub) AddValidator(validate Validator) error {
 // Subscribe provides a new Subscription for EDS notifications.
 func (s *PubSub) Subscribe() (*Subscription, error) {
 	if s.topic == nil {
-		return nil, fmt.Errorf("share/eds: topic is not started")
+		return nil, fmt.Errorf("share/shrexpush: topic is not started")
```
Comment (Wondertan) [inline share/p2p/shrexpush/pubsub.go:84]: nit, but this seems to be enough
```suggestion
		return nil, fmt.Errorf("shrexpush: topic is not started")
```

File: share/p2p/shrexpush/pubsub.go [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
+	logging "github.com/ipfs/go-log/v2"
+
 	"github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/peer"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 
 	"github.com/celestiaorg/celestia-node/share"
 )
 
+var log = logging.Logger("share/shrexpush")
```
Comment (Wondertan) [inline share/p2p/shrexpush/pubsub.go:16]: Nit:
```suggestion
var log = logging.Logger("shrexpush")
```

**Review by Wondertan — APPROVED** [review]
shrexpush we may want to remove share prefixes. Just protocol name is descriptive enough

**Review by distractedm1nd — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**